crates/network/src/encode.rs

```diff
@@ -0,0 +1,52 @@
+use bytes::{Buf, BufMut, BytesMut};
+
+use rjacraft_protocol::{Encoder, VarInt};
+
+#[derive(Default)]
+pub struct PacketEncoder {
+    buf: BytesMut,
+}
+
+impl PacketEncoder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn append_frame(&mut self, bytes: &[u8]) -> anyhow::Result<()> {
+        let data_len = VarInt(bytes.len() as i32);
+        let start_len = self.buf.len();
+        let data_len_size = data_len.written_size();
+
+        self.buf.put_bytes(0, data_len_size);
+        let mut front = &mut self.buf[start_len..];
+        data_len.encode_write(&mut front)?;
```

Comment:
There is no need for the local variable and it feels like the code is easier to read without it:

```suggestion
        data_len.encode_write(&mut self.buf[start_len..])?;
```